doc/adr/0006-explicit-resources-for-declaring-service-bindings.md

```diff
@@ -0,0 +1,68 @@
+# 6. Explicit resources for declaring service bindings
+
+Date: 2024-01-16
+
+## Status
+
+Proposed
+
+Enables [7. Deprecate ServiceSync](0007-deprecate-servicesync.md)
```

Comment:
I might be clear to state that the existing `skupper.io/proxy` annotation
is also going to be deprecated by this change.